simulator/bpu/bpu.h

```diff
@@ -67,11 +73,13 @@ class BP : public BaseBP
 
     Addr get_target( Addr PC) final
     {
-        uint32 way;
-        bool is_hit;
-        std::tie( is_hit, way) = tags.read_no_touch( PC);
-        if ( is_hit) // hit
+        // return saved target only in case it is predicted taken
```

Comment:
Previous code was more clear. There was no std::ignore, first of all. Please revert to the previous version.

Reply:
The previous code retrieved the target even when predictor return NOT TAKEN. In current BPU implementation the target does not alter when state goes from T to NT and vice versa, so it is another bug. Probably it is better to be resolved some other way (which, then?)

Reply:
Slightly changed the logic so as not to use `std::ignore`. It introduces double-check of `is_hit`, unfortunately.

Reply:
The number 1 goal of this project is clear code, understandable by students who just read Schildt's "Teach Yourself C++".

Reply:
Of course, I get it.
Is this part OK now?